Core/Src/bmbInterface.c

```diff
@@ -369,28 +334,26 @@ static bool loadAndVerifyTxQueue(uint8_t *data_p, uint32_t numBytes)
 
 		// Write queue
 		csOn();
-		// Ensure semaphore set to 0
-		xSemaphoreTake(asciSpiSemHandle, 0);
-		SPIRTRY(HAL_SPI_Transmit_IT, &hspi1, data_p, numBytes);
-		if (xSemaphoreTake(asciSpiSemHandle, TIMEOUT_SPI_COMPLETE_MS) != pdTRUE)
+		INTERRUPT_STATUS_E spiStatus = SPI_TRANSMIT(HAL_SPI_Transmit_IT, &hspi1, TIMEOUT_SPI_COMPLETE_MS, loadAndVerifyTxQueue, data_p, numBytes);
```

Comment:
Why is spiStatus an interrupt status? Bit confusing can this be tidied up at all?